include/interface/blas3_interface.hpp

```diff
@@ -133,20 +129,31 @@ cl::sycl::event _gemm(Executor& ex, char _TransA, char _TransB, IndexType _M,
         _ldc);                                                             \
   }
 #ifndef NAIVE_GEMM
+#if defined(DYNAMIC)
   if (ex.get_device_type() ==
       Executor::Queue_Interface_Type::device_type::SYCL_INTEL_GPU) {
     BIND_DATA_SIZE(1024, 4096, 1024) TO_TPARAMS(128, false, 4, 4, 16, 16);
     BIND_DATA_SIZE(10, 1024, 1024) TO_TPARAMS(128, false, 2, 2, 8, 8);
     BIND_DEFAULT TO_TPARAMS(128, false, 8, 8, 8, 8);
   } else if ((ex.get_device_type() == Executor::Queue_Interface_Type::
-                                          device_type::SYCL_RCAR_CVENGINE) &&
+                                          device_type::SYCL_RCAR_CVENGINE) ||
              (ex.get_device_type() == Executor::Queue_Interface_Type::
                                           device_type::SYCL_RCAR_HOST_CPU)) {
-    BIND_DEFAULT TO_TPARAMS(32, false, 8, 8, 8, 8);
+    BIND_DEFAULT TO_TPARAMS(16, false, 4, 4, 8, 4);
   } else {
     BIND_DATA_SIZE(10, 1024, 1024) TO_TPARAMS(128, true, 1, 1, 16, 16);
     BIND_DEFAULT TO_TPARAMS(128, false, 8, 8, 16, 16);
   }
+#elif defined(INTEL_GPU)
+  BIND_DATA_SIZE(1024, 4096, 1024) TO_TPARAMS(128, false, 4, 4, 16, 16);
+  BIND_DATA_SIZE(10, 1024, 1024) TO_TPARAMS(128, false, 2, 2, 8, 8);
+  BIND_DEFAULT TO_TPARAMS(128, false, 8, 8, 8, 8);
+#elif defined(RCAR)
+  BIND_DEFAULT TO_TPARAMS(16, false, 4, 4, 8, 4);
+#else  // any other specified devices
+  BIND_DATA_SIZE(10, 1024, 1024) TO_TPARAMS(128, true, 1, 1, 16, 16);
+  BIND_DEFAULT TO_TPARAMS(128, false, 8, 8, 16, 16);
```

Comment:
can you add some comments explaining all this numbers? Is easy to lose track of all the different parameters.